js/framework/widgets/PDFView.js

```diff
@@ -391,9 +391,16 @@ var PDFView = new Knowledge.Class({
     },
 
     vfunc_get_preferred_width: function () {
-        let width = this.get_screen().get_width();
         let [minimal, ] = this.parent();
-        return [minimal, width];
+        let natural = null;
+
+        let gdk_window = this.get_window();
+        let gdk_display = gdk_window.get_display();
+        let gdk_monitor = gdk_display.get_monitor_at_window(gdk_window);
```

Comment:
You can remove `gdk_display` and the code is still readable.

Reply:
Okay, done :) I realized that gdk_monitor should always be set, as well, so I removed the condition there.